packages/knip/src/reporters/gitlab.ts

```diff
@@ -0,0 +1,128 @@
+import { createHash } from 'node:crypto';
+import type { Entries } from 'type-fest';
+import type { Issue, IssueRecords, IssueSeverity, IssueSymbol, Report, ReporterOptions } from '../types/issues.js';
+import { getTitle } from './util.js';
+
+/**
+ * Entry for GitLab Code Quality report. See: https://docs.gitlab.com/ci/testing/code_quality/#code-quality-report-format
+ */
+interface CodeQualityEntry {
+  check_name: string;
+  description: string;
+  severity: 'info' | 'minor' | 'major' | 'critical' | 'blocker';
+  fingerprint: string;
+  location: {
+    path: string;
+    lines: {
+      begin: number;
+    };
+  };
+}
+
+export default async ({ report, issues }: ReporterOptions) => {
+  const entries: CodeQualityEntry[] = [];
+  const hashes = new Set<string>();
+
+  for (const [type, isReportType] of Object.entries(report) as Entries<Report>) {
+    if (!isReportType) {
+      continue;
+    }
+
+    if (type === 'files' || type === '_files') {
+      continue;
+    }
```

Comment:
Maybe we could leave in the unused files type?

I know it's ugly, but it'd need something like this:

```ts
type = type === 'files' ? '_files' : type;
```

And ignore `files`. No worries if you don't want/need it now, I could work it in later on.

Reply:
What is the difference between "files" and "_files"?

Reply:
`files` is a Set of file paths, but I've added `_files` later on so that all issue types have the same object structure for consistency/ease of use (and I can wrap this up a in next major with a breaking change).